### ISSUE TITLE
Missing results: Same torrent from multiple providers

### ISSUE THREAD

**Issue by burekas7** [elgatito/script.elementum.burst#157]
### The Problem:

When I'm running a search for this episode for example:
The Witcher S01E02

It seems that The Pirate Bay and GIOTorrents has the same torrent.
But instead of presenting that it's existing in both of them.
When they both checked the results shows only the result from GIOTorrents and ignoring the result from The Pirate Bay which is preferable because it has many more seeds and peers.

### Expected:
I'm expecting that it will show that the same torrent exist in different providers,
and when selecting it, it will start downloading from the providers that has (therotical) the best seeds/peers count.

### Images & Logs

* With The Pirate Bay & Without GIOTorrents
Kodi Log: https://paste.ubuntu.com/p/XNCyH34YtV
![image](https://user-images.githubusercontent.com/13424257/71549934-bed73c80-29ce-11ea-8ec8-d1ae6a842fd1.png)

* With The Pirate Bay & With GIOTorrents:
It ignore the the same result from The Pirate Bay and show only the result from GIOTorrents as shown above.
Kodi Log: https://paste.ubuntu.com/p/8H2zvdZdZg
![image](https://user-images.githubusercontent.com/13424257/71550067-c4ce1d00-29d0-11ea-9175-7739317e0aff.png)

In addition, when I add the "Rutor" provider, it ignores almost all of the "The Pirate Bay" results from the results list, why?
Kodi Log in this case: https://paste.ubuntu.com/p/gvY6NsZj57

---

Elementum: 0.1.43
Burst: 0.0.40
Kodi: 17.6
Windows 10, 64bit

**Comment by elgatito** [comment]
> I'm expecting that it will show that the same torrent exist in different providers

To union torrents - they have to have same ID (Hash). 
Hash is the checksum of files, so if hash differs - files are different or there are more files in the torrent.

**Comment by burekas7** [comment]
Ok I get it.
1) So assuming they are different because what you said.
Why when I add GIOTorrent for example, it ignore this specific result that has better ratio from The Pirate Bay and show it just from GIOTorrent.

2) What is the algorithem of presenting the results? By their ratio, a few from each provider ignoring the ratio, something else...?
* It relates also to the seconde issue I mentioned here when I add the Rutor provider.

**Comment by elgatito** [comment]
```
00:01:41.096 T:20948  NOTICE: [script.elementum.burst] >>             KickAss returned 38 results in 33.8 seconds, sending 20 best ones
```

In the log you can see messages like this.
Sorting is configured in Burst settings.
By default it's 
```
            sort_balance = sort_seeds * 3 * sort_resolution
```

The bigger resolution, the bigger - balance is.

**Comment by burekas7** [comment]
Ok, even by this rule the result from The Pirate Bay should be there, they are the same resolution and it has better seeds, and when I remove GIOTorrents, this specific result is in the second place from The Pirate Bay. So I don't understand why it doesn't include it in the results when GIOTorrents exist?

And even when I set it to be sorted by seeds, this result isn't show up.
Only the result from GIOTorrent and not from The Pirate Bay which is better as mentioned.

I'm expecting to get the best results, and here it hidden better result then the other regarding to this specific torrent, which is the same (but different as you said)

Log: By seeds and without GIOTorrents:
https://paste.ubuntu.com/p/gwfKtgsCB6

Log: By seeds and with GIOTorrents:
https://paste.ubuntu.com/p/XZrBc5JRmd

**Comment by elgatito** [comment]
Burst is filtering for EACH provider, not in overall.
Is result is filtered - then it's filtered in favor of other results.

**Comment by elgatito** [comment]
Your logs say nothing without debug mode.

**Comment by burekas7** [comment]
Log: By seeds and without GIOTorrents:
https://paste.ubuntu.com/p/K47S8qqx5k

Log: By seeds and with GIOTorrents:
https://paste.ubuntu.com/p/swWjnqk4T8

I set also max result for each provider to 25
And sort by seeds.
And the same, when GIOtorrents is activate I don't see this torrent from The Pirate, why?

Elementum 0.1.43
Burst 0.0.40

**Comment by elgatito** [comment]
```
22:54:28.135 T:12680   DEBUG: [script.elementum.burst] [torlock]                 the witcher us s01 complete 1080p web x264-strife[tgx] *** Name mismatch
```

Try to disable Name/Title filtering in Burst settings. See if that helps.

**Comment by burekas7** [comment]
What do you mean?
Which of them?

![image](https://user-images.githubusercontent.com/13424257/71601561-a09f4700-2b5c-11ea-8d4e-bf59b1b32523.png)

**Comment by burekas7** [comment]
> ```
> 22:54:28.135 T:12680   DEBUG: [script.elementum.burst] [torlock]                 the witcher us s01 complete 1080p web x264-strife[tgx] *** Name mismatch
> ```
> 
> Try to disable Name/Title filtering in Burst settings. See if that helps.

By the way, I mentioned it regarding The Pirate Bay & GIOTorrents and not TorLock

**Comment by elgatito** [comment]
> Which of them?

Required keywords.

> By the way, I mentioned it regarding The Pirate Bay & GIOTorrents and not TorLock

I don't see parser messages for thepiratebay.
Maybe you send too much requests for pages. Try to disable few providers and see if you see thepiratebay back.

**Comment by burekas7** [comment]
It seems to be better in 0.0.44 (with the same filters)

According to this comment:
https://github.com/elgatito/script.elementum.burst/issues/142#issuecomment-569951444

This time it preffered the TPB result while the GLO isn't included in the results screen.
Generally it's better. But I need to do more testing to verify the stability.

What is the general number of results I can see from each provider?
Because I set it to the maximum, so why I never see this specific result from both of them?

Kodi Log:
https://paste.ubuntu.com/p/jGgwy2RCg9

**Comment by burekas7** [comment]
It's still not so stable, I can see this specific result in the results screen only from one provider at a time, sometimes it can be only from TPB and sometimes only from GLO, but never together, why? 
(Even they have the same name, you said they are should act as different torrents) 

When the torrent is shown only from TPB (Which has better seeds/peers count):
Log: https://paste.ubuntu.com/p/fS352zCgcq

When the torrent is shown only from GLO:
Log: https://paste.ubuntu.com/p/njVFGwgbpG

**Comment by burekas7** [comment]
Updated tests for this issue:
I installed a clean kodi 17.6 (Elementum 0.1.51, Burst 0.0.48, Without touching the settings)
Made a manual search for: Money Heist S04E03.
And I got this:
Kodi Log: https://paste.ubuntu.com/p/QKx3WfVDcd
![image](https://user-images.githubusercontent.com/13424257/80310585-32a21e00-87e4-11ea-9080-f2ef31444f92.png)


And also make the same search with my current kodi, and got this:
Kodi log: https://paste.ubuntu.com/p/pGmHJfvk9f
![image](https://user-images.githubusercontent.com/13424257/80310546-f078dc80-87e3-11ea-8ef2-9d238534827a.png)


So my question is: Why I don't get this result for example: Money-Heist-S04E03-720p-WEB-x264-GHOSTS-eztv from 1337x in my current build (or the rest results) while in the clean build I do get them.

**Comment by burekas7** [comment]
It seems that it is fixed on Elementum Burst 0.0.57
More details here: elgatito/script.elementum.burst#224
